### PR TITLE
Development

### PR DESCRIPTION
**Pull request by gjbex**
## Summary by Sourcery

Add new CMake examples demonstrating FetchContent, Google Test integration, and CMake presets.

New Features:
- Introduce a FetchContent-based example that downloads and uses the CLI11 library in a random number generator utility.
- Add a Google Test example project showcasing unit testing of a Fibonacci function with CTest integration.
- Provide a CMake presets example project illustrating debug and release builds with a simple executable and optional test.

Documentation:
- Extend top-level example READMEs to reference the new FetchContent, Google Test, and CMake presets sample projects.
- Document usage and structure of the new FetchContent example, including CLI11-based argument parsing.
- Document usage and structure of the Google Test + CTest example and the CMake presets example, including basic build and test commands.

Tests:
- Add a Google Test suite for the Fibonacci function covering negative inputs, base cases, and several recursive values.

### CODE REVIEW

File: source_code/presets/CMakePresets.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "version": 8,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 20,
+    "patch": 0
```
[the remark this answers is not in the document]
Comment (gjbex) [reply source_code/presets/CMakePresets.json]: @copilot apply changes based on [this feedback](https://github.com/gjbex/CMake-usecases/pull/5#discussion_r3026967935)

File: source_code/presets/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(hello LANGUAGES CXX)
```
[the remark this answers is not in the document]
Comment (gjbex) [reply source_code/presets/CMakeLists.txt]: @copilot apply changes based on [this feedback](https://github.com/gjbex/CMake-usecases/pull/5#discussion_r3026967904)

File: source_code/fetch_content/random_default.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    int n {default_n};
+    double a {default_a};
+    double b {default_b};
+    size_t seed;
+    CLI::App app{"Generate random numbers"};
+    app.add_option("-n", n, "number of random values to generate")->default_val(std::to_string(default_n));
+    app.add_option("-a", a, "minimum value")->default_val(std::to_string(default_a));
+    app.add_option("-b", b, "maximum value")->default_val(std::to_string(default_b));
+    app.add_option("--seed", seed, "seed to use");
+    try {
```
[the remark this answers is not in the document]
Comment (gjbex) [reply source_code/fetch_content/random_default.cpp:37]: @copilot apply changes based on [this feedback](https://github.com/gjbex/CMake-usecases/pull/5#discussion_r3026967963)

I would prefer a default value for `seed`.

File: source_code/presets/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+two presets: `debug` and `release`, which can be used to build the project in
+debug or release mode, respectively.
+
+
+## What is it?
+
+1. `CMakePresets.json` defines the presets for configuring and building the project.
+1. `src/main.cpp` is the source code for the project, which simply prints
+   "Hello, World!" to the console.
+1. `CMakeLists.txt` is the CMake configuration file that defines how to build the
```
[the remark this answers is not in the document]
Comment (gjbex) [reply source_code/presets/README.md:13]: @copilot apply changes based on [this feedback](https://github.com/gjbex/CMake-usecases/pull/5#discussion_r3026967851)

File: source_code/ctest/google_test/fib.cpp [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#include <stdexcept>
+
+int fib(int n) {
+    if (n < 0) {
```
[the remark this answers is not in the document]
Comment (gjbex) [reply source_code/ctest/google_test/fib.cpp:5]: @copilot apply changes based on [this feedback](https://github.com/gjbex/CMake-usecases/pull/5#discussion_r3026968016)

File: source_code/presets/CMakePresets.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
+{
+  "version": 8,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 20,
+    "patch": 0
+  },
```
[the remark this answers is not in the document]
Comment (gjbex) [reply source_code/presets/CMakePresets.json]: @copilot apply changes based on [this feedback](https://github.com/gjbex/CMake-usecases/pull/5#discussion_r3026968027)

Please raise the CMake mininum version.
